include/ddc/kernels/splines/splines_linear_problem_2x2_blocks.hpp

```diff
@@ -0,0 +1,260 @@
+     *
+     * So we perform the factorization inplace to store only the relevant blocks in the matrix (while factorizing
+     * the blocks themselves if necessary):
+     *
+     * |   Q    |         Q^-1*gamma        |
+     * | lambda | delta - lambda*Q^-1*gamma |
+     */
+    void setup_solver() override
+    {
+        // Setup the top-left solver
+        m_top_left_block->setup_solver();
+
+        // Compute Q^-1*gamma in top-right block
+        m_top_right_block.modify_host();
+        m_top_right_block.sync_device();
+        m_top_left_block->solve(m_top_right_block.d_view);
+        m_top_right_block.modify_device();
+        m_top_right_block.sync_host();
+
+        // Push lambda on device in bottom-left block
+        m_bottom_left_block.modify_host();
+        m_bottom_left_block.sync_device();
+
+        // Compute delta - lambda*Q^-1*gamma in bottom-right block
+        compute_schur_complement();
+
+        // Setup the bottom-right solver
+        m_bottom_right_block->setup_solver();
+    }
+
+    /**
+     * @brief Compute y <- y - LinOp*x or y <- y - LinOp^t*x.
+     *
+     * [SHOULD BE PRIVATE (GPU programming limitation)]
+     *
+     * @param x
+     * @param y
+     * @param LinOp
+     * @param transpose
+     */
+    void gemv_minus1_1(
+            MultiRHS const x,
+            MultiRHS const y,
+            Kokkos::View<double**, Kokkos::LayoutRight, typename ExecSpace::memory_space> const
+                    LinOp,
+            bool const transpose = false) const
+    {
+        assert(!transpose && LinOp.extent(0) == y.extent(0)
+               || transpose && LinOp.extent(1) == y.extent(0));
+        assert(!transpose && LinOp.extent(1) == x.extent(0)
+               || transpose && LinOp.extent(0) == x.extent(0));
+        assert(x.extent(1) == y.extent(1));
+
+        Kokkos::parallel_for(
+                "gemv_minus1_1",
+                Kokkos::TeamPolicy<ExecSpace>(y.extent(0) * y.extent(1), Kokkos::AUTO),
+                KOKKOS_LAMBDA(
+                        const typename Kokkos::TeamPolicy<ExecSpace>::member_type& teamMember) {
+                    const int i = teamMember.league_rank() / y.extent(1);
+                    const int j = teamMember.league_rank() % y.extent(1);
+
+                    double LinOpTimesX = 0.;
+                    Kokkos::parallel_reduce(
+                            Kokkos::TeamThreadRange(teamMember, x.extent(0)),
+                            [&](const int l, double& y_tmp) {
+                                if (!transpose) {
+                                    y_tmp += LinOp(i, l) * x(l, j);
+                                } else {
+                                    y_tmp += LinOp(l, i) * x(l, j);
+                                }
+                            },
+                            LinOpTimesX);
+                    if (teamMember.team_rank() == 0) {
+                        y(i, j) -= LinOpTimesX;
+                    }
```

Comment:
I tried to avoid this by writing ie. `y_tmp -= LinOp(i, l) * x(l, j);` but it does not work.

Reply:
Yeah, I tried:

```
                    Kokkos::parallel_reduce(
                            Kokkos::TeamThreadRange(teamMember, x.extent(0)),
                            [&](const int l, double& y_tmp) {
                                if (!transpose) {
                                    y_tmp -= LinOp(i, l) * x(l, j);
                                } else {
                                    y_tmp -= LinOp(l, i) * x(l, j);
                                }
                            },
                            y(i,j));
 ```
 
 With y being of course not initialized to 0.

Reply:
If i remember correctly, `parallel_reduce` does not contribute to the variable result, `y(i, j)` in the above code, it simply overwrites it.